### PR TITLE
Lint entire codebase

### PR DESCRIPTION
**Pull request by joeizang**
Give the whole project a once over manually linting every file.

### CODE REVIEW

**Comment by bunsenstraat** [comment]
are the rules used in linting this also in the lint config? meaning, does it checked during the lint step in CI? otherwise it wouldn't make sense.

**Comment by joeizang** [comment]
> are the rules used in linting this also in the lint config? meaning, does it checked during the lint step in CI? otherwise it wouldn't make sense.

basically yes. It follows the eslint rules you created. I just applied them per file manually.

File: apps/circuit-compiler/webpack.config.js [hunk 1]
```diff
@@ -58,13 +57,12 @@ module.exports = composePlugins(withNx(), (config) => {
   // souce-map loader
   config.module.rules.push({
     test: /\.js$/,
-    use: ["source-map-loader"],
-    enforce: "pre"
+    use: ['source-map-loader'],
+    enforce: 'pre',
```
[the remark this answers is not in the document]
Comment (joeizang) [reply apps/circuit-compiler/webpack.config.js:61]: This isn't a json file. its a js file

File: apps/doc-gen/src/app/docgen/common/properties.ts [hunk 2]
```diff
@@ -36,47 +36,47 @@ export function fullName ({ item, contract }: DocItemContext): string {
...
-    case 'VariableDeclaration':
-      return formatVariable(item);
+  case 'ErrorDefinition': {
+    const params = item.parameters.parameters;
+    return `error ${item.name}(${params.map(formatVariable).join(', ')})`;
+  }
+
+  case 'ModifierDefinition': {
+    const params = item.parameters.parameters;
+    return `modifier ${item.name}(${params.map(formatVariable).join(', ')})`;
```
[the remark this answers is not in the document]
Comment (joeizang) [reply apps/doc-gen/src/app/docgen/common/properties.ts]: resolved

**Comment by bunsenstraat** [comment]
> > are the rules used in linting this also in the lint config? meaning, does it checked during the lint step in CI? otherwise it wouldn't make sense.
> 
> basically yes. It follows the eslint rules you created. I just applied them per file manually.

then why would linting pass in the previous codebase? if the rule was in place.. aren't we mixing prettier and linting.

**Comment by joeizang** [comment]
> > > are the rules used in linting this also in the lint config? meaning, does it checked during the lint step in CI? otherwise it wouldn't make sense.
> > 
> > 
> > basically yes. It follows the eslint rules you created. I just applied them per file manually.
> 
> then why would linting pass in the previous codebase? if the rule was in place.. aren't we mixing prettier and linting.

@bunsenstraat I think we aren't. The thing is prettier ensures the files are formatted with 2 spaces with 2 or 3 rules for semi colons, & commas. So I just did the 2 space thing for code files that were missed (somehow) and fix any existing semi colons.

File: apps/remix-ide/src/assets/js/loader.js [hunk 3]
```diff
@@ -15,40 +15,40 @@ const domainsSecondaryTracker = {
...
-    var u = "https://ethereumfoundation.matomo.cloud/";
-    _paq.push(['setTrackerUrl', u + 'matomo.php']);
-    _paq.push(['setSiteId', domains[window.location.hostname]]);
-    var d = document, g = d.createElement('script'), s = d.getElementsByTagName('script')[0];
-    g.async = true; g.src = '//cdn.matomo.cloud/ethereumfoundation.matomo.cloud/matomo.js'; s.parentNode.insertBefore(g,s);
+    var u = "https://ethereumfoundation.matomo.cloud/"
+    _paq.push(['setTrackerUrl', u + 'matomo.php'])
+    _paq.push(['setSiteId', domains[window.location.hostname]])
+    var d = document, g = d.createElement('script'), s = d.getElementsByTagName('script')[0]
+    g.async = true g.src = '//cdn.matomo.cloud/ethereumfoundation.matomo.cloud/matomo.js' s.parentNode.insertBefore(g,s)
```
Comment (bunsenstraat) [inline apps/remix-ide/src/assets/js/loader.js:34]: see this is an example of where problems come with this method...

**Comment by bunsenstraat** [comment]
What I suggest we do is:
we runt lint fix on each nx project and leave the other files untouched. 
and we sync the prettier file to reflect the rules exactly as we discussed earlier. 
I don't feel confident merging it like this. the removal ; fix breaks some things if applied carelessly.
for example:
console.log('test')
(window as any).doSomething()
would cause errors with a ; in between.
also errors happened with:  
g.async = true; g.src = '//cdn.matomo.cloud/ethereumfoundation.matomo.cloud/matomo.js' s.parentNode.insertBefore(g,s)
g.async = true g.src = '//cdn.matomo.cloud/ethereumfoundation.matomo.cloud/matomo.js' s.parentNode.insertBefore(g,s)
so this method of fixing is not reliable.

**Pull request closed without merging**